### DISCUSSION TITLE
if field match regex

### DISCUSSION THREAD

**Discussion by filetto** [vectordotdev/vector#20262] [Q&A] [answered]
I'd like to do the following:

`regex_public_ip = r'^10\.10\.10\.(?:[0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$'`

`if .source_ip != regex_public_ip {`

`do something`

`}`

So the problem is, this ifnot don't catch the regex but only like a hard coded private IP address for testing purposes. I need something like if not match, but I dont know how to wirte. Any ideas?

The application is OpenObserve and gives me the following identifiers to choose from:

`expected one of: "\n", "!=", "&&", "*", "+", "-", "/", "<", "<=", "==", ">", ">=", "??", "{", "|", "|", "||"`

**Comment by jszwedko** [comment, marked as answer]
I think you want to use https://vector.dev/docs/reference/vrl/functions/#match to check if the input matches a regex (rather than `!=`).

**Reply by filetto** [reply]
Thank you very much! Didn't know how to negate that, ! before match did the trick!
`if (!match(value:.source_ip, pattern:regex_public_ip)) { do something }`
